server/src/main/java/org/elasticsearch/health/node/DiskHealthIndicatorService.java

```diff
@@ -489,4 +149,264 @@ private HealthIndicatorDetails getDetails(boolean explain, Map<String, DiskHealt
+                        Locale.ROOT,
+                        " Furthermore %d node%s with roles: [%s] %s out of disk or running low on disk space.",
+                        unhealthyNodesCount,
+                        unhealthyNodesCount == 1 ? "" : "s",
+                        roles,
+                        unhealthyNodesCount == 1 ? "is" : "are"
+                    );
+                }
+            } else {
+                String roles = getSortedUniqueValuesString(affectedRoles, DiscoveryNodeRole::roleName);
+                int unhealthyNodesCount = masterNodes.size() + otherNodes.size() + dataNodes.size();
+                symptom = String.format(
+                    Locale.ROOT,
+                    "%d node%s with roles: [%s] %s out of disk or running low on disk space.",
+                    unhealthyNodesCount,
+                    unhealthyNodesCount == 1 ? "" : "s",
+                    roles,
+                    unhealthyNodesCount == 1 ? "is" : "are"
+                );
+            }
+            return symptom;
+        }
+
+        List<HealthIndicatorImpact> getImpacts() {
+            if (healthStatus == HealthStatus.GREEN) {
+                return List.of();
+            }
+            List<HealthIndicatorImpact> impacts = new ArrayList<>();
+            if (hasBlockedIndices()) {
+                impacts.add(
+                    new HealthIndicatorImpact(
+                        NAME,
+                        IMPACT_INGEST_UNAVAILABLE_ID,
+                        1,
+                        String.format(
+                            Locale.ROOT,
+                            "Cannot insert or update documents in the affected indices [%s].",
+                            getTruncatedIndices(blockedIndices, clusterState.getMetadata())
+                        ),
+                        List.of(ImpactArea.INGEST)
+                    )
+                );
+            } else {
+                if (indicesAtRisk.isEmpty() == false) {
+                    impacts.add(
+                        new HealthIndicatorImpact(
+                            NAME,
+                            IMPACT_INGEST_AT_RISK_ID,
+                            1,
+                            String.format(
+                                Locale.ROOT,
+                                "The cluster is at risk of not being able to insert or update documents in the affected indices [%s].",
+                                getTruncatedIndices(indicesAtRisk, clusterState.metadata())
+                            ),
+                            List.of(ImpactArea.INGEST)
+                        )
+                    );
+                }
+            }
+            if (affectedRoles.contains(DiscoveryNodeRole.MASTER_ROLE)) {
+                impacts.add(
+                    new HealthIndicatorImpact(
+                        NAME,
+                        IMPACT_CLUSTER_STABILITY_AT_RISK_ID,
+                        2,
```

Comment:
I think this should also be severity 1

Reply:
Hm, no strong feelings. I will change it.